tests/flakes/flake-in-submodule.sh

```diff
@@ -0,0 +1,54 @@
+source common.sh
+
+# Tests that:
+# - flake.nix may reside inside of a git submodule
+# - the flake can access content outside of the submodule
+#
+#   rootRepo
+#   ├── root.nix
+#   └── submodule
+#       ├── flake.nix
+#       └── sub.nix
+
+
+requireGit
+
+clearStore
+
+# Submodules can't be fetched locally by default.
+# See fetchGitSubmodules.sh
+export XDG_CONFIG_HOME=$TEST_HOME/.config
+git config --global protocol.file.allow always
+
+
+rootRepo=$TEST_ROOT/rootRepo
+subRepo=$TEST_ROOT/submodule
+
+rm -rf $rootRepo $subRepo $TEST_HOME/.cache/nix
+
+
+createGitRepo $subRepo
+cat > $subRepo/flake.nix <<EOF
+{
+    outputs = { self }: {
+        sub = import ./sub.nix;
+        root = import ../root.nix;
+    };
+}
+EOF
+echo '"expression in submodule"' > $subRepo/sub.nix
+git -C $subRepo add flake.nix sub.nix
+git -C $subRepo commit -m Initial
+
+createGitRepo $rootRepo
+
+git -C $rootRepo submodule init
+git -C $rootRepo submodule add $subRepo submodule
+echo '"expression in root repo"' > $rootRepo/root.nix
+git -C $rootRepo add root.nix
+git -C $rootRepo commit -m "Add root.nix"
+
+# Flake can live inside a submodule and can be accessed via ?dir=submodule
+[[ $(nix eval --json git+file://$rootRepo\?submodules=1\&dir=submodule#sub ) = '"expression in submodule"' ]]
+# The flake can access content outside of the submodule
+[[ $(nix eval --json git+file://$rootRepo\?submodules=1\&dir=submodule#root ) = '"expression in root repo"' ]]
```

Comment:
The evaluation cache will work better when this is disallowed by default.

(The design is still experimental. As a community member I try my best to help finalize it, and I appreciate your help.)

Reply:
> The evaluation cache will work better when this is disallowed by default.

Thank you for the insight. While I was aware that changes in this regard are currently under discussion, I was not aware that this was due to the evaluation cache.

> If we change the behavior before releasing flakes as stable, this test case will help provide a solution for your use case.

Thank you for considering (and merging)! I know this might be a somewhat odd use case here, but it reflects a situation of mine, where the company I work for has (for managing access control) dozens of repos that (for reproducibility) get tracked and auto-updated as git-submodules of a central git repository. Having the possibility to keep all nix expressions in one submodule, while being able to reference and build the code of the "sibling" repositories is what this test here should guarantee. Thank you for making it possible to have early feedback on potentially upcoming changes in that regard. ♥️

Reply:
> I was not aware that this was due to the evaluation cache.

I can't claim to know about all the discussions, plural, but this would be one consideration.

yw!